### PR TITLE
Fix issues with missing monadfail constraints

### PR DESCRIPTION
**Pull request by fredefox**
This is an error in GHC >= 8.6

### CODE REVIEW

File: src/Snap/Extras/SpliceUtils/Interpreted.hs [hunk 1]
```diff
@@ -146,7 +148,9 @@ scriptsSplice d prefix = do
 -- > beta-functions-enabled = true
 ifFlagSplice :: SnapletISplice b
 ifFlagSplice = do
-  Element _ ats es <- getParamNode
+  e <- getParamNode
+  -- TODO Incomplete pattern matching!
```
Comment (MichaelXavier) [inline src/Snap/Extras/SpliceUtils/Interpreted.hs]: What needs to happen here? Was this always incomplete and we're just hearing about it now?

Comment (fredefox) [reply src/Snap/Extras/SpliceUtils/Interpreted.hs]: fail used to be an instance of the Monad type class but cannot be implemented sensibly for all monads, so they must resolve to using error for those cases. This is similar to my transformation, the error will just be reported as an incomplete pattern match. I don't remember the specific context at this line, but whatever monad it is doesn't have a monadfail instance so is not allowed to have fallible pattern matches. Hope that makes sense.

Comment (MichaelXavier) [reply src/Snap/Extras/SpliceUtils/Interpreted.hs]: That makes sense. Sounds like this change makes things no worse.

**Comment by MichaelXavier** [comment]
Thanks! Seems like I should hold on releasing this because none of the major upstream packages like snap seem to have resolutions for the nightlies, and the latest LTS is still on 8.4.3

**Comment by MichaelXavier** [comment]
Fair enough. This is released with `snap-extras-0.12.2.1`. Thanks again!

**Comment by mightybyte** [comment]
I just pushed different commit which I anticipate will be the approach I take to solve this issue.  You can find it here:

https://github.com/snapframework/snap/commit/5ba5df1d14541044e9e23d145b853e672018640f

Can you give that a try and let me know if this fix works?

**Pull request merged**